pkg/server/registry/apigroups/acorn/apps/validator.go

```diff
@@ -135,11 +135,14 @@ func (s *Validator) validateRegion(ctx context.Context, app *apiv1.App) error {
 	}
 
 	if app.Spec.Region == "" {
-		if project.Spec.DefaultRegion == "" && project.Status.DefaultRegion == "" {
-			return fmt.Errorf("no region can be determined because project default region is not set")
+		app.Status.Defaults.Region = project.Spec.DefaultRegion
```

Comment:
We have to have something to indicate the default region of an app so that it gets synced properly.

Answer to your side question: yes. If the project default changes, then the app default will change. However, this should be handled correctly. That switch should be rare, however. I don't want apps to be switching regions constantly and possible cause issues.

Reply:
Technically speaking, an app's region is immutable. If a user supplies a region when they run an app, then they cannot change it.

The default is not immutable. I did this to be consistent with other scenarios. An admin changes a default on a project and eventually all the things that use that default are changed, too.

I'd be happy to change this to be immutable as well if we can come up with reasonable behavior.